internal/codec/measurement_test.go

```diff
@@ -0,0 +1,97 @@
+package codec
+
+import (
+	"io"
+	"sync"
+	"testing"
+	"time"
+)
+
+func TestMeasureBitRateStatic(t *testing.T) {
+	r, w := io.Pipe()
+	dur := time.Second * 5
+	dataSize := 1000
+	var precision float64 = 8 // 1 byte
+	var wg sync.WaitGroup
+	wg.Add(1)
+	done := make(chan struct{})
+
+	go func() {
+		data := make([]byte, dataSize)
+		// Make sure that this goroutine is synchronized with main goroutine
+		wg.Done()
+		ticker := time.NewTicker(time.Second)
+
+		for {
+			select {
+			case <-ticker.C:
+				w.Write(data)
+			case <-done:
+				w.Close()
+				break
```

Comment:
It breaks inner select block and continue outer for loop.
```suggestion
				return
```

Reply:
ouch, my bad...